src/librustc_typeck/impl_wf_check.rs

```diff
@@ -99,6 +99,10 @@ fn enforce_impl_params_are_constrained(
 ) {
     // Every lifetime used in an associated type must be constrained.
     let impl_self_ty = tcx.type_of(impl_def_id);
+    if impl_self_ty.sty == ty::Error {
```

Comment:
Please don't use `==` on `.sty` like this (you can compare `impl_self_ty` with `tcx.types.err` if you want, but nothing does that really).

I'm *pretty sure* what you want is more like `impl_self_ty.references_error()`, but I'd ask @nikomatsakis for a quick look to be sure.

Reply:
I've seen `.references_error()` used in similar places so I know that is correct.